src/models/stores/stores.ts

```diff
@@ -128,23 +132,48 @@ export const setUnitAndProblem = async (stores: IStores, unitId: string | undefi
   stores.unit = UnitModel.create(unitJson);
   const {investigation, problem} = stores.unit.getProblem(_problemOrdinal);
 
+  // TODO: make this dynamic like the way the components work. The components
+  // access these values from the stores when they need them. This way the values
+  // can be changed on the fly without having to track down each object that is
+  // using them.
   stores.documents.setAppConfig(stores.appConfig);
+  stores.documents.setFirestore(stores.db.firestore);
+  stores.documents.setUserContext(getUserContext(stores));
+
   if (investigation && problem) {
     stores.investigation = investigation;
     stores.problem = problem;
   }
   stores.problemPath = getProblemPath(stores);
 
-  // need to use a listener because user type can be determined after unit initialization
-  unit.installUserListener(() => stores.user.isTeacher, async (isTeacher: boolean) => {
-    // only load the teacher guide content for teachers
-    if (isTeacher) {
-      const guideJson = await getGuideJson(unitId, stores.appConfig);
-      const unitGuide = guideJson && UnitModel.create(guideJson);
-      const teacherGuide = unitGuide?.getProblem(problemOrdinal || stores.appConfig.defaultProblemOrdinal)?.problem;
-      stores.teacherGuide = teacherGuide;
-    }
-  });
+  // need to use a listener because user type, id, name can be determined after unit initialization
+  addDisposer(unit, reaction(() => ({
+      isTeacher: stores.user.isTeacher,
+      userContext: getUserContext(stores)
+    }),
+    async ({isTeacher, userContext}: {isTeacher: boolean, userContext: IUserContext}) => {
```

Comment:
The `userContext` contains the type of user, so you could eliminate `isTeacher` from the access function:
```suggestion
  addDisposer(unit, reaction(() => getUserContext(stores),
    async (userContext: IUserContext) => {
```